tests/ui/associated-types/hr-associated-type-bound-param-3.stderr

```diff
@@ -15,10 +15,10 @@ LL |     for<'b> <T as X<'b, T>>::U: Clone,
    |                                 ^^^^^ required by this bound in `X`
 
 error[E0277]: the trait bound `str: Clone` is not satisfied
-  --> $DIR/hr-associated-type-bound-param-3.rs:18:5
+  --> $DIR/hr-associated-type-bound-param-3.rs:18:18
    |
 LL |     <(i32,) as X<(i32,)>>::f("abc");
```

Comment:
This is a great example of the bug. We have `X<'a, T>` above, but have only specified the type parameter. Since we were basically trying to index at `1` into a list of `1`, we would fail. Now we properly adjust the index to account for the lifetime not being there :D